# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/BindGroupLayout.cpp
# Commit: dd10bf41fcea
# Full Hash: dd10bf41fcea6aa2f40d4528f4a7afc22b1f389e
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2024-04-05 21:43:02
# Regressor Bug: 1881518
# File Overlap Count: 1
# Description:
#   Bug 1881518 - Recycle indices again. r=webgpu-reviewers,bradwerth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202433
# ==============================================================================

diff -r b2cba928a7bb -r dd10bf41fcea dom/webgpu/BindGroupLayout.cpp
--- a/dom/webgpu/BindGroupLayout.cpp	Fri Apr 05 07:28:07 2024 +0000
+++ b/dom/webgpu/BindGroupLayout.cpp	Fri Apr 05 07:42:05 2024 +0000
@@ -22,12 +22,22 @@
 BindGroupLayout::~BindGroupLayout() { Cleanup(); }
 
 void BindGroupLayout::Cleanup() {
-  if (mValid && mParent) {
-    mValid = false;
-    auto bridge = mParent->GetBridge();
-    if (mOwning && bridge && bridge->IsOpen()) {
+  if (!mValid) {
+    return;
+  }
+  mValid = false;
+
+  auto bridge = mParent->GetBridge();
+  if (!bridge) {
+    return;
+  }
+
+  if (mOwning) {
+    if (bridge->IsOpen()) {
       bridge->SendBindGroupLayoutDrop(mId);
     }
+
+    wgpu_client_free_bind_group_layout_id(bridge->GetClient(), mId);
   }
 }
 